### ISSUE TITLE
ladvdc not working correctly on Raspberry Pi

### ISSUE THREAD

**Issue by schoerg** [sspans/ladvd#24]
Hi,

I am using ladvd, compiled from source on a Raspberry Pi.
`Linux raspberrypi 4.1.13-v7+ #826 SMP PREEMPT Fri Nov 13 20:19:03 GMT 2015 armv7l GNU/Linux`

The daemon is working as intended, its sending LLDP PDUs as expected. On other x86 based machines, I can view the LLDP table with the populated devices, including the Raspberry Pi.

Then I have another Raspberry Pi, also running ladvd. I try to view the LLDP neighbors by running ladvdc, but the output is "stuck" at this point:

```
$ ladvdc
Capability Codes:
    r - Repeater, B - Bridge, H - Host, R - Router, S - Switch,
    W - WLAN Access Point, C - DOCSIS Device, T - Telephone, O - Other

Device ID           Local Intf    Proto   Hold-time    Capability    Port ID

```

I then have to close ladvdc with Ctrl+C. However LLDP PDUs arrive at that device, as I confirmed with tcpdump. I can share the dumps if you like.

**Comment by sspans** [comment]
My apologies for not getting to you sooner.

Can you run ladvd with more debugging to figure out what's going on?
And yes, dumps would be helpful. But this most likely is a ladvd internal problem.
Running the process with strace would be useful as well.

**Comment by schoerg** [comment]
`[22:09:17] root@rpi ~ # ladvd -fvvvv
ladvd 0.9.2 running
fetching all interfaces
skipping interface lo
found ethernet interface eth0
adding interface eth0
detecting eth0 settings
pcilib: Cannot open /proc/bus/pci
pcilib: Cannot find any working access method.
`
I hope this is correct.

**Comment by sspans** [comment]
This looks like an issue with pcilib / libpci. Can you try recompiling without libpci?

**Comment by sspans** [comment]
Can you try if 	b6d5964 fixes this for you?

**Comment by schoerg** [comment]
Sorry for the late comment, but yes the problem is fixed with b6d5964.
